### PR TITLE
Allow SHA or Tag to be supplied for installer

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Tested on a vagrant box and it seems to work.

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1452290452022118

**Comment by manasdk** [comment]
nice!

**Comment by enykeev** [comment]
I'm probably a little bit slow this morning, how is it different from `-b [ref]` besides the fact it creates a new branch? Why do we need that branch?

File: script/bootstrap-st2 [hunk 1]
```diff
@@ -14,6 +14,7 @@ show_help() {
   echo "Usage: $0 [OPTION]..."
   echo
   echo "  -a,  --answers     Supply installer with an answer file"
+  echo "  -r,  -- ref        Supply a SHA or TAG to use as installer version"
```
Comment (dzimine) [inline script/bootstrap-st2]: while with that, add -b help?

Comment (lakshmi-kannan) [inline script/bootstrap-st2]: I left it out intentionally. It appears currently this feature is not a customer facing one. So let's not document it. Customers should only use -r option.

**Comment by lakshmi-kannan** [comment]
> I'm probably a little bit slow this morning, how is it different from -b [ref] besides the fact it creates a new branch? Why do we need that branch?

I am confused. -b support was added by you IIRC. It's done so we can do testing in st2cd. I think -b suppport should stay. -r is now added and will be a customer facing option so they can switch to a particular tag/sha of workroom if they need to. Tags are how st2workroom will be versioned.

**Comment by enykeev** [comment]
What I'm trying to say is that they have basically the same functionality short of renaming the branch. If this renaming is not required and simply made for convenience, lets pick one and remove the other. Having two is confusing no matter whether we going to document `-b` or not.

**Comment by lakshmi-kannan** [comment]
> If this renaming is not required and simply made for convenience,

For tags and sha, to make sure we don't get a detached HEAD state which might throw off some users, I decided to add -b some_branch_name. I see what you mean w.r.t having one as opposed to two options. I'll remove -b then :P. We need to update the st2cd workflows as well. I'll open a PR for that.

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1452574040022538

**Comment by dzimine** [comment]
The default should ideally be resent stable version. So `curl -sSL https://install.stackstorm.com/ | sudo sh` installs the latest stable version (1.3 once we release). Currently the default is 'master' 

Discussed with @lakshmi-kannan and we keep it at it for now but need to revisit.

**Comment by Kami** [comment]
Hm, I don't know if my issue is related to this change, but just in case.

`ENV=<branch name>` doesn't seem to work for my anymore. Am I missing something?

**Pull request merged**